go/libraries/doltcore/sqle/dprocedures/dolt_rebase.go

```diff
@@ -524,58 +736,103 @@ func processRebasePlanStep(ctx *sql.Context, planStep *rebase.RebasePlanStep,
 		}
 	}
 
+	// If the action is "drop", then we don't need to do anything
+	if planStep.Action == rebase.RebaseActionDrop {
+		return nil
+	}
+
+	options, err := createCherryPickOptionsForRebaseStep(ctx, planStep, commitBecomesEmptyHandling, emptyCommitHandling)
+	if err != nil {
+		return err
+	}
+
+	return handleRebaseCherryPick(ctx, planStep, *options)
+}
+
+func createCherryPickOptionsForRebaseStep(ctx *sql.Context, planStep *rebase.RebasePlanStep, commitBecomesEmptyHandling doltdb.EmptyCommitHandling, emptyCommitHandling doltdb.EmptyCommitHandling) (*cherry_pick.CherryPickOptions, error) {
 	// Override the default empty commit handling options for cherry-pick, since
 	// rebase has slightly different defaults
 	options := cherry_pick.NewCherryPickOptions()
 	options.CommitBecomesEmptyHandling = commitBecomesEmptyHandling
 	options.EmptyCommitHandling = emptyCommitHandling
 
 	switch planStep.Action {
-	case rebase.RebaseActionDrop:
-		return nil
+	case rebase.RebaseActionDrop, rebase.RebaseActionPick:
```

Comment:
Yeah, that reads a little funny... I updated the comment to explain that drop and pick don't require any special options because the drop action is a no-op, and the pick action doesn't need to amend or use a custom commit message, so the basic cherry-pick options are enough for it.